JDK11 causes compile error

Hi, I'm trying to migrate some projects from JDK 8 to JDK 11/10 and get a compilation error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project mapstruct-jdk11: Compilation failure
[ERROR] Couldn't find type java.sql.Date. Are you missing a dependency on your classpath?
```
I build on:
```
➜  mapstruct-jdk11 git:(master) mvn --version
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)
Maven home: /home/alfred/Workspaces/Apps/apache-maven
Java version: 11.0.4, vendor: Azul Systems, Inc., runtime: /home/alfred/Workspaces/Apps/jdk_zulu11.33.15-ca-fx-jdk11.0.4-linux_x64
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-55-generic", arch: "amd64", family: "unix"
```
You can reproduce the error with:
https://github.com/alfredalbrecht/mapstruct-jdk11

Thanks and Regards.

--- a/integrationtest/src/test/java/org/mapstruct/itest/tests/MavenIntegrationTest.java
+++ b/integrationtest/src/test/java/org/mapstruct/itest/tests/MavenIntegrationTest.java
@@ -130,4 +130,10 @@ public class MavenIntegrationTest {
     void targetTypeGenerationTest() {
     }
 
+    @ProcessorTest(baseDir = "moduleInfoTest")
+    @EnabledForJreRange(min = JRE.JAVA_11)
+    void moduleInfoTest() {
+
+    }
+
 }

--- a/integrationtest/src/test/resources/moduleInfoTest/pom.xml
+++ b/integrationtest/src/test/resources/moduleInfoTest/pom.xml
@@ -1,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.mapstruct</groupId>
+        <artifactId>mapstruct-it-parent</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>modulesTest</artifactId>
+    <packaging>jar</packaging>
+
+</project>

--- a/integrationtest/src/test/resources/moduleInfoTest/src/main/java/module-info.java
+++ b/integrationtest/src/test/resources/moduleInfoTest/src/main/java/module-info.java
@@ -1,0 +1,9 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+module test.mapstruct {
+    requires org.mapstruct;
+    exports org.mapstruct.itest.modules;
+}

--- a/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerDto.java
+++ b/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerDto.java
@@ -1,0 +1,31 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.modules;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class CustomerDto {
+
+    private String name;
+    private String email;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+}

--- a/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerEntity.java
+++ b/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerEntity.java
@@ -1,0 +1,31 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.modules;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class CustomerEntity {
+
+    private String name;
+    private String mail;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getMail() {
+        return mail;
+    }
+
+    public void setMail(String mail) {
+        this.mail = mail;
+    }
+}

--- a/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerMapper.java
+++ b/integrationtest/src/test/resources/moduleInfoTest/src/main/java/org/mapstruct/itest/modules/CustomerMapper.java
@@ -1,0 +1,23 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.modules;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface CustomerMapper {
+
+    CustomerMapper INSTANCE = Mappers.getMapper( CustomerMapper.class );
+
+    @Mapping(target = "mail", source = "email")
+    CustomerEntity fromDto(CustomerDto record);
+
+}

--- a/integrationtest/src/test/resources/moduleInfoTest/src/test/java/org/mapstruct/itest/modules/ModulesTest.java
+++ b/integrationtest/src/test/resources/moduleInfoTest/src/test/java/org/mapstruct/itest/modules/ModulesTest.java
@@ -1,0 +1,28 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.modules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+import org.mapstruct.itest.modules.CustomerDto;
+import org.mapstruct.itest.modules.CustomerEntity;
+import org.mapstruct.itest.modules.CustomerMapper;
+
+public class ModulesTest {
+
+    @Test
+    public void shouldMapRecord() {
+        CustomerDto dto = new CustomerDto();
+        dto.setName( "Kermit" );
+        dto.setEmail( "kermit@test.com" );
+        CustomerEntity customer = CustomerMapper.INSTANCE.fromDto( dto );
+
+        assertThat( customer ).isNotNull();
+        assertThat( customer.getName() ).isEqualTo( "Kermit" );
+        assertThat( customer.getMail() ).isEqualTo( "kermit@test.com" );
+    }
+}

--- a/processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java
+++ b/processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java
@@ -184,9 +184,8 @@ public class Conversions {
         register( Enum.class, String.class, new EnumStringConversion() );
         register( Date.class, String.class, new DateToStringConversion() );
         register( BigDecimal.class, BigInteger.class, new BigDecimalToBigIntegerConversion() );
-        register( Date.class, Time.class, new DateToSqlTimeConversion() );
-        register( Date.class, java.sql.Date.class, new DateToSqlDateConversion() );
-        register( Date.class, Timestamp.class, new DateToSqlTimestampConversion() );
+
+        registerJavaTimeSqlConversions();
 
         // java.util.Currency <~> String
         register( Currency.class, String.class, new CurrencyToStringConversion() );
@@ -227,13 +226,26 @@ public class Conversions {
         register( ZonedDateTime.class, Date.class, new JavaZonedDateTimeToDateConversion() );
         register( LocalDateTime.class, Date.class, new JavaLocalDateTimeToDateConversion() );
         register( LocalDate.class, Date.class, new JavaLocalDateToDateConversion() );
-        register( LocalDate.class, java.sql.Date.class, new JavaLocalDateToSqlDateConversion() );
         register( Instant.class, Date.class, new JavaInstantToDateConversion() );
 
     }
 
+    private void registerJavaTimeSqlConversions() {
+        if ( isJavaSqlAvailable() ) {
+            register( LocalDate.class, java.sql.Date.class, new JavaLocalDateToSqlDateConversion() );
+
+            register( Date.class, Time.class, new DateToSqlTimeConversion() );
+            register( Date.class, java.sql.Date.class, new DateToSqlDateConversion() );
+            register( Date.class, Timestamp.class, new DateToSqlTimestampConversion() );
+        }
+    }
+
     private boolean isJodaTimeAvailable() {
         return typeFactory.isTypeAvailable( JodaTimeConstants.DATE_TIME_FQN );
+    }
+
+    private boolean isJavaSqlAvailable() {
+        return typeFactory.isTypeAvailable( "java.sql.Date" );
     }
 
     private void registerNativeTypeConversion(Class<?> sourceType, Class<?> targetType) {

